prov/shm/src/smr_cntr.c

```diff
@@ -32,16 +32,30 @@
 
 #include "smr.h"
 
+static struct fi_ops_cntr smr_cntr_ops = {
+	.size = sizeof(struct fi_ops_cntr),
+	.read = ofi_cntr_read,
+	.readerr = ofi_cntr_readerr,
+	.add = ofi_cntr_add,
+	.adderr = ofi_cntr_adderr,
+	.set = ofi_cntr_set,
+	.seterr = ofi_cntr_seterr,
+	.wait = ofi_cntr_spin_wait
+};
+
 int smr_cntr_open(struct fid_domain *domain, struct fi_cntr_attr *attr,
 		  struct fid_cntr **cntr_fid, void *context)
 {
 	int ret;
 	struct util_cntr *cntr;
 
-	if (attr->wait_obj != FI_WAIT_NONE) {
-		FI_INFO(&smr_prov, FI_LOG_CNTR, "cntr wait not yet supported\n");
+	if (attr->wait_obj != FI_WAIT_NONE &&
+	    attr->wait_obj != FI_WAIT_UNSPEC) {
+		FI_INFO(&smr_prov, FI_LOG_CNTR,
+			"cntr wait objects not yet supported\n");
 		return -FI_ENOSYS;
 	}
+	attr->wait_obj = FI_WAIT_NONE;
```

Comment:
This implies no waiting is supported.  It's okay to leave as UNSPEC if that's what the app requessted.

Reply:
I interpreted WAIT_NONE as no waiting object. I changed it to WAIT_NONE mainly to prevent the utility code from opening a wait object which it does for UNSPEC. Thoughts?

Reply:
While typing my last reply, the thought occurred to me that instead of adding spinning cntr and cq calls, we could instead modify the wait set to support WAIT_NONE.  Calling fi_wait() on a WAIT_NONE wait set would just return 0.

If we do this, I think the existing cq/cntr sread calls might work.  And if that's the case, then the existing cq/cntr signal calls might work as well.

Reply:
Along these lines, maybe can consider exposing FI_WAIT_YIELD directly in the API, mostly for output purposes, so that the app can see that waiting results in spinning a thread on the CPU, with calls to yield.

Reply:
That is a wonderful idea. Very smart of you. Good boy. Necco wafers for you!